pheme/transformation/scanreport/gvmd.py

```diff
@@ -0,0 +1,205 @@
+) -> Tuple[int, Optional[Union[HostResults, NVTResult]]]:
+    for i, candidate in enumerate(candidates):
+        if is_the_chosen_one(candidate):
+            return i, candidate
+    return -1, None
+
+
+def group_by_host(first, second):
+    host = second.pop('host')
+    host = host if isinstance(host, str) else host["text"]
+    index, hr = __find_first_occurance(first, lambda h: h.host == host)
+    # hr = first.get(host)
+    nvt = second.pop('nvt')
+    solution = (
+        Solution(nvt['solution']['type'], nvt['solution']['text'])
+        if nvt.get('solution')
+        else None
+    )
+    refs = (
+        [Ref(r['id'], r['type']) for r in nvt['refs']['ref']]
+        if nvt.get('refs')
+        else []
+    )
+    oid = nvt['oid']
+    qod = (
+        QOD(second['qod']['value'], second['qod']['type'],)
+        if second.get('qod')
+        else None
+    )
+    shr = HostResult(
+        oid,
+        nvt.get('type'),
+        nvt.get('name'),
+        nvt.get('family'),
+        nvt.get('cvss_base'),
+        nvt.get('tags'),
+        solution,
+        refs,
+        second.get('port'),
+        second.get('threat'),
+        second.get('severity'),
+        qod,
+        second.get('description'),
+    )
+    if hr:
+        hr.results.append(shr)
+        first[index] = hr
+    # first[host] = hr
+    else:
+        # first[host] = HostResults(host, [shr])
+        first.append(HostResults(host, [shr]))
+    return first
+
+
+def group_by_nvt(first, second):
+    nvt = second.pop('nvt')
+    host = second.pop('host')
+    host = host if isinstance(host, str) else host["text"]
+    oid = nvt['oid']
+    index, nvt_result = __find_first_occurance(first, lambda n: n.oid == oid)
+    hs = HostSpecific(host, second.get('description'))
+    if nvt_result:
+        nvt_result.hosts.append(hs)
+        first[index] = nvt_result
+    else:
+        solution = (
+            Solution(nvt['solution']['type'], nvt['solution']['text'])
+            if nvt.get('solution')
+            else None
+        )
+        refs = (
+            [Ref(r['id'], r['type']) for r in nvt['refs']['ref']]
+            if nvt.get('refs')
+            else []
+        )
+        qod = (
+            QOD(second['qod']['value'], second['qod']['type'],)
+            if second.get('qod')
+            else None
+        )
+        first.append(
+            NVTResult(
+                oid,
+                nvt.get('type'),
+                nvt.get('name'),
+                nvt.get('family'),
+                nvt['cvss_base'],
+                nvt.get('tags'),
+                solution,
+                refs,
+                second.get('port'),
+                second.get('threat'),
+                second.get('severity'),
+                qod,
+                [hs],
+            )
+        )
+    return first
+
+
+def transform(data: dict, group_by: Callable = group_by_host) -> Report:
```

Comment:
Would be nice to know which type of dict this is here (hint needs an additional import `from typing import Dict`).
```suggestion
def transform(data: Dict[str, str], group_by: Callable = group_by_host) -> Report:
```